test/Microsoft.AspNetCore.Diagnostics.Tests/ExceptionHandlerTest.cs

```diff
@@ -494,5 +494,26 @@ public class ExceptionHandlerTest
             Assert.NotNull(listener.DiagnosticHandledException?.HttpContext);
             Assert.NotNull(listener.DiagnosticHandledException?.Exception);
         }
+
+        [Fact]
+        public void UsingExceptionHandler_ThrowsAnException_WhenExceptionHandlingPathNotSet()
+        {
+            // Arrange
+            DiagnosticListener diagnosticListener = null;
+
+            var builder = new WebHostBuilder()
+                .Configure(app =>
+                {
+                    diagnosticListener = app.ApplicationServices.GetRequiredService<DiagnosticListener>();
+
+                    // Act
+                    var exception = Assert.Throws<InvalidOperationException>(() => app.UseExceptionHandler());
```

Comment:
The exception won't be thrown until you call builder.Build().

Reply:
No, the exception is thrown from the ctor. Unless that isn't what should happen